ua_box_api/ua_box_api.py

```diff
@@ -9,69 +8,84 @@
+            private_key_passphrase=stache_secret["boxAppSettings"]["appAuth"][
+                "passphrase"
+            ],
+        )
+        auth = BoxJWTAuth(config=jwt_config)
+        self.client = BoxClient(auth=auth)
 
-    def get_all_items(self, item_id):
+    def get_all_items(self, item_id: int | str):
         """Returns list of all items in the object with the given item_id."""
         # If a folder is passed in, it gets caught in an infinite while loop
         # with a bare except somewhere -- instead, check that the id is an int.
         if not (isinstance(item_id, int) or isinstance(item_id, str)):
             raise TypeError("Item_id must be an int.")
-        folder = self.client.folder(folder_id=item_id)
 
-        items = list()
-        offset = 0
-        has_next_item = True
+        item_id = str(item_id)
 
-        # Every 300000 items, get a new generator; otherwise, add the current
-        # generator's next(). If the current generator doesn't have a next(),
-        # make while condition False.
-        while has_next_item:
-            if len(items) == offset:
-                items_generator = folder.get_items(
-                    limit=offset + 300000, offset=offset)
-                offset += 300000
-            try:
-                items.append(items_generator.next())
-            except StopIteration:
-                has_next_item = False
+        folder = self.client.folders.get_folder_items(folder_id=item_id, usemarker=True)
+        items = folder.entries
+        next_item = folder.next_marker
+        while next_item is not None:
+            folder = self.client.folders.get_folder_items(
+                folder_id=item_id, usemarker=True, marker=next_item
+            )
+            if folder.entries is not None:
+                items += folder.entries
+                next_item = folder.next_marker
+            else:
+                break
 
-        return items
+        if items is None:
+            raise FileNotFoundError("Failed to find requested folder.")
+        else:
+            return items
 
-    def find_child_by_name(self, name, item_id):
+    def find_child_by_name(self, name: str, item_id: int | str):
         """Returns object with name if found in item_id, or None if not."""
-        matches = [x for x in self.get_all_items(item_id) if x.name == name]
-        if matches:
-            return matches[0]
-        return None
+        search = self.client.search.search_for_content(
+            query=f'"{name}"',  # Wrap query in double quotes, because Box API.
+            ancestor_folder_ids=[str(item_id)],
+            content_types=["name"],
+        )
 
-    def get_duplicate_file_name(self, folder_id, current_name, zip_file=False):
+        # This construct throws a ValueError if there isn't exactly 1 match.
+        # https://stackoverflow.com/a/7008062
+        # https://dbader.org/blog/python-nested-unpacking
```

Comment:
This kind of documentation is distracting. Should we be returning a collection of matches? If you need to add stackoverflow links, I'd suggest using ADO for that.